libs/api/Verifier.cpp

```diff
@@ -773,9 +769,14 @@ static _Success_(return == 0) uint32_t _verify_section_from_string(
     }
 
     auto stream = std::stringstream(data);
-    struct _thread_local_storage_cache tls_cache;
+
+    // Clear thread local storage before calling into the verifier.
+    // Note that TLS should be cleared here *before* calling into the verifier, not after.
+    // Post verification, bpf2c relies on the TLS cache to compute program info hash.
+    ebpf_clear_thread_local_storage();
+
     set_global_program_and_attach_type(program_type, nullptr);
-    set_verification_in_progress(true);
+    _verification_in_progress_helper helper;
```

Comment:
This change is to ensure that `_verification_in_progress` is reset to `false` when this function returns.